### PR TITLE
[JENKINS-57965] Prevent save of KubernetesSlave through UI

### PR DESCRIPTION
**Pull request by Vlatombe**

### CODE REVIEW

**Comment by jglick** [comment]
Not sure if test timeouts are flakes or not. I did notice this which might be related:

```
java.lang.NullPointerException
	at hudson.model.Node.toComputer(Node.java:208)
	at hudson.model.Slave.getComputer(Slave.java:539)
	at org.csanchez.jenkins.plugins.kubernetes.KubernetesLauncher.launch(KubernetesLauncher.java:144)
	at hudson.slaves.SlaveComputer$1.call(SlaveComputer.java:294)
	at …
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java [hunk 1]
```diff
@@ -195,6 +197,12 @@ public KubernetesComputer createComputer() {
         return KubernetesComputerFactory.createInstance(this);
     }
 
+    @Override
+    public Node reconfigure(StaplerRequest req, JSONObject form) throws Descriptor.FormException {
+        // This is not meant to be reconfigured through UI.
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java]: Maybe better to just override `KubernetesComputer.doConfigSubmit` to throw an exception? Or override `KubernetesSlave.getACL` to return one which denies `CONFIGURE` to everyone (but permits `VIEW_CONFIGURATION` where applicable), like for example `WorkflowJob` does when it is a branch project?

Comment (Vlatombe) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java]: ah right, something based on `VIEW_CONFIGURATION` would be nicer

Comment (jglick) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java]: That would leave a `configure` screen intact and merely remove the **Save** / **Apply** buttons. Also should be honored by `doConfigSubmit`.

**Comment by Vlatombe** [comment]
I looked at the failure, and what I understood is the following:
* Jenkins is being shut down
* The Kubernetes launcher was still running and throws a NPE because Jenkins singleton is now null. Ideally we'd need to interrupt the launcher thread upon shutdown?
* The timeout seems to be caused by a serie of remoting calls which execute serially (each thread is trying to lock the Queue), and the one holding the lock is waiting on a remoting call.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java [hunk 2]
```diff
@@ -145,4 +144,16 @@ public String toString() {
...
 
+    @Override
+    public ACL getACL() {
+        final ACL base = super.getACL();
+        return new ACL() {
+            @Override
+            public boolean hasPermission(Authentication a, Permission permission) {
+                if (permission== Computer.CONFIGURE)
+                    return false;
+                return base.hasPermission(a,permission);
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java]: nonstandard formatting these three lines, FYI

**Pull request merged**